### ISSUE TITLE
Refactor and clean up backend API and models

### ISSUE THREAD

**Issue by ekaltman** [shftgroup/gisst#172]
The GISST models and routes files are rather large, unwieldy, and hard to parse or browse. I'm putting this issue in as a marker to clean that up at some point. I'm also closing other sub-issues related to this issue for the time being. 

We probably want to either adopt a frame for DB access, like Diesel, or just break the model / routes into separate files based on context. This might also be a job for macros if I get the time. 

Ideally, we'd have the different API endpoint routes and models in their own files, so that its easier to go in and locate things to change or modify. There is soooo much duplicated boilerplate across the route and model definitions right now.

**Comment by JoeOsborn** [comment]
I pared it back some in #186 , and moved to only using concrete types instead of traits.  I don't know that an issue is the best way to address this, and probably some of the "generic query actions" in model.rs should be replaced by stuff more specific to our actual needs (like, gisst-cli is a bit inconvenient to use; we could make it more convenient and share more code with ingest by refactoring both into gisst where possible, like I did with insert_file_object recently).
